GitExtUtils/LazyStringSplit.cs

```diff
@@ -0,0 +1,173 @@
+        private readonly char _delimiter;
+        private readonly StringSplitOptions _options;
+
+        public LazyStringSplit(string input, char delimiter, StringSplitOptions options = StringSplitOptions.None)
+        {
+            Requires.NotNull(input, nameof(input));
+
+            _input = input;
+            _delimiter = delimiter;
+            _options = options;
+        }
+
+        public Enumerator GetEnumerator() => new(this, _options);
+
+        IEnumerator IEnumerable.GetEnumerator() => GetEnumerator();
+
+        IEnumerator<string> IEnumerable<string>.GetEnumerator() => GetEnumerator();
+
+        public struct Enumerator : IEnumerator<string>
+        {
+            private readonly StringSplitOptions _options;
+            private readonly string _input;
+            private readonly char _delimiter;
+            private int _index;
+
+            internal Enumerator(in LazyStringSplit split, StringSplitOptions options)
+            {
+                _options = options;
+                _index = 0;
+                _input = split._input;
+                _delimiter = split._delimiter;
+                Current = null!;
+            }
+
+            public string Current { get; private set; }
+
+            public bool MoveNext()
+            {
+                while (_index < _input.Length)
+                {
+                    int delimiterIndex = _input.IndexOf(_delimiter, _index);
+
+                    if (delimiterIndex == -1)
+                    {
+                        Current = _input.Substring(_index);
+                        _index = _input.Length + 1;
+                        return true;
+                    }
+
+                    int length = delimiterIndex - _index;
+
+                    if (length == 0)
+                    {
+                        _index++;
+                        if (_options == StringSplitOptions.RemoveEmptyEntries)
+                        {
+                            continue;
+                        }
+                        else
+                        {
+                            Current = "";
+                            return true;
+                        }
+                    }
+
+                    Current = _input.Substring(_index, length);
+                    _index = delimiterIndex + 1;
+
+                    return true;
+                }
+
+                if (_options == StringSplitOptions.None && _index == _input.Length)
+                {
+                    _index++;
+                    Current = "";
+                    return true;
+                }
+
+                return false;
+            }
+
+            object IEnumerator.Current => Current;
+
+            void IEnumerator.Reset()
+            {
+                _index = 0;
+                Current = null!;
+            }
+
+            void IDisposable.Dispose()
+            {
+            }
+        }
+    }
+
+    internal static class LazyStringSplitExtensions
```

Comment:
Own file please

Reply:
This was intentional. The file is linked into multiple projects. Splitting it out will create more linked files, and create the opportunity for future usages to only link one file, where they then wouldn't benefit from these optimised replacements for common methods that would otherwise allocate.

Reply:
Alright, leave it as is.